src/Pact/Analyze/Types/Shared.hs

```diff
@@ -482,6 +482,26 @@ newtype VarId
   = VarId Int
   deriving (Show, Eq, Enum, Num, Ord)
 
+-- | Identifier name that is guaranteed to be unique because it contains a
+-- unique identifier. These names are generated upstream in the compiler.
```

Comment:
where do we generate these names?

Reply:
The typechecker, using `freshId` there. I will update this comment.